recipes/backward-cpp/all/conanfile.py

```diff
@@ -0,0 +1,152 @@
+       "stack_details_backtrace_symbol": [True, False],
+       "stack_details_dw": [True, False],
+       "stack_details_bfd": [True, False],
+       "stack_details_dwarf": [True, False],
+       "shared": [True, False],
+       "fPIC": [True, False]
+    }
+    default_options = {
+       "stack_walking_unwind": True,
+       "stack_walking_backtrace": False,
+       "stack_details_auto_detect": False,
+       "stack_details_dw": False,
+       "stack_details_bfd": False,
+       "stack_details_dwarf": True,
+       "stack_details_backtrace_symbol": False,
+       "shared": True,
+       "fPIC": True
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+
+    def config_options(self):     
+          if self.settings.os == "Macos":
+            del self.options.stack_details_dw
+            del self.options.stack_details_bfd
+            del self.options.stack_details_dwarf
+    
+    def _check_options(self, options):
+        sum = 0
+        for name, value in self.options.values.as_list():
+            if value and name in options:
+                sum = sum + 1
+        if sum != 1:
+            return False
+        return True
+
+    def configure(self):
+        if self.settings.os not in ["Linux", "Macos", "Android"]:
+            raise ConanInvalidConfiguration("upstream backward-cpp v{0} is not \
+                supported in {1}.".format(self.version, self.settings.os))
+        # windows implementation only available in upstream master branch
+        
+        if not self._check_options(['stack_walking_unwind', 'stack_walking_backtrace']):
+            raise ConanInvalidConfiguration("Please select stack_walking_unwind"
+                                            " or stack_walking_backtrace.")
+
+        if self.settings.os in ["Linux", "Android"]:
+            if not self._check_options(['stack_details_dw', \
+                                        'stack_details_bfd', \
+                                        'stack_details_dwarf', \
+                                        'stack_details_backtrace_symbol']):
+                raise ConanInvalidConfiguration("Please select stack_details_dw"\
+                    ", stack_details_bfd, stack_details_dwarf or " \
+                    "stack_details_backtrace_symbol.")
+    
+        if self.settings.os == "Macos":            
+            if not self._check_options(['stack_details_backtrace_symbol']):
+                raise ConanInvalidConfiguration("Please select stack_details_backtrace_symbol.")
+
+    def requirements(self):
+        if self.settings.os in ["Linux", "Android"] and \
+           self.options.stack_details_dwarf:
+            self.requires("libdwarf/20191104")
+    
+    def system_requirements(self):
+        required_package = None
+        if self.settings.os == "Linux":
+            if self.options.stack_details_dw:
+                required_package = "libdw-dev"
```

Comment:
Is this a library that we should create a conan recipe for?

Reply:
actually, we have libdwarf already available in CCI: https://github.com/conan-io/conan-center-index/pull/409

Reply:
actually, we already have it in CCI: https://github.com/conan-io/conan-center-index/pull/409

Reply:
Probably yes. But libdw is a component of `elfutils` https://sourceware.org/elfutils/ should libdw be packaged separately or as a component of an hypotetical elfutils package?

Reply:
@uilianries Yes, CCI libwarf is used when options ask for libdwarf (it's used as the default option).

Reply:
> But libdw is a component of elfutils

I was missing this part, thanks for the clarification. So yes, Bincrafters has a package for elfutils: https://github.com/bincrafters/conan-elfutils It can absorbed by CCI

Reply:
I've checked the bincrafters/conan-elfutils and I have some doubts. If i'm not wrong, it adds all libraries (`libelf`, `libdw` & `libasm`) `cpp_info.libs`? That means that a consumer can't out-out to linking some of those libs.
For instance, if backward-cpp specifies a dependency with conan-elfutils, I would like to link against libdw only.
For that we require https://github.com/conan-io/conan/pull/5408 right?

Reply:
Yes, conan-io/conan#5408 should solve it, but also, we can provide libdw package, the second option is will be faster.

Reply:
I'll look to package libdw then.

Reply:
It would be great! :+1:

Reply:
Going back to the idea of packaging `elfutils-libdw`. It turns out it requires `elfutils-libelf`. I guess this would require modifications in the elfutils autotools to be able to compile libelf only, package it, and then compile libdw without the internal libelf but with the provided libelf by conan.
I'm thinking if it's not a better idea to restrict `stack_details` to `dwarf & backtrace_symbol`.